### PR TITLE
cask audit: check for allow_untrusted within caskroom taps

### PR DESCRIPTION
**Pull request by commitay**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
cc @vitorgalvao @reitermarkus

### CODE REVIEW

File: Library/Homebrew/cask/lib/hbc/audit.rb [hunk 1]
```diff
@@ -50,6 +51,18 @@ def summary_header
 
     private
 
+    def check_untrusted_pkg
+      odebug "Auditing pkg stanza: allow_untrusted"
+
+      return if @cask.sourcefile_path.nil?
+
+      tap = Tap.select { |t| t.cask_file?(@cask.sourcefile_path) }.first
+      return if tap.nil? || tap.user != "caskroom"
```
Comment (reitermarkus) [inline Library/Homebrew/cask/lib/hbc/audit.rb]: Does `@cask.tap` work here instead of `Tap.select`?

File: Library/Homebrew/cask/lib/hbc/audit.rb [hunk 2]
```diff
@@ -50,6 +51,18 @@ def summary_header
...
+
+      return unless cask.artifacts.count { |k| k.is_a?(Hbc::Artifact::Pkg) && k.stanza_options.key?(:allow_untrusted) }.positive?
```
Comment (reitermarkus) [inline Library/Homebrew/cask/lib/hbc/audit.rb]: Instead of `count {}.positive?` use `any? {}`.

File: Library/Homebrew/test/cask/audit_spec.rb [hunk 3]
```diff
@@ -91,6 +91,25 @@ def include_msg?(messages, msg)
...
+        let(:cask_token) { "basic-cask" }
+        it { should_not warn_with(error_msg) }
+      end
+
+      context "when the cask does not have allow_untrusted" do
+        let(:cask_token) { "with-uninstall-pkgutil" }
+        it { should_not warn_with(error_msg) }
+      end
+
+      context "when the cask has allow_untrusted" do
```
Comment (reitermarkus) [inline Library/Homebrew/test/cask/audit_spec.rb]: Uppercase “Cask” in these.

File: Library/Homebrew/cask/lib/hbc/audit.rb (hunk 1 above)
Comment (commitay) [reply Library/Homebrew/cask/lib/hbc/audit.rb]: This is used by another check.

https://github.com/Homebrew/brew/blob/6034caf9fb8e2b7ac86f97f2cf6c3ced621c8920/Library/Homebrew/cask/lib/hbc/audit.rb#L95-L99

File: Library/Homebrew/test/cask/audit_spec.rb (hunk 3 above)
Comment (commitay) [reply Library/Homebrew/test/cask/audit_spec.rb]: Lowercase is consistent with the other checks.

File: Library/Homebrew/cask/lib/hbc/audit.rb (hunk 1 above)
Comment (commitay) [reply Library/Homebrew/cask/lib/hbc/audit.rb]: `@cask.tap` works, should I change both?

Comment (reitermarkus) [reply Library/Homebrew/cask/lib/hbc/audit.rb]: Yes, change both.

File: Library/Homebrew/test/cask/audit_spec.rb (hunk 3 above)
Comment (reitermarkus) [reply Library/Homebrew/test/cask/audit_spec.rb]: I generally try to capitalize Homebrew terminology like Tap, Formula, Cask etc. so they stand out a bit.

**Pull request merged**